Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -204,27 +209,32 @@ struct CombinatorialKalmanFilterOptions {
   LoggerWrapper logger;
 };
 
+template <typename traj_t>
 struct CombinatorialKalmanFilterResult {
   // Fitted states that the actor has handled.
-  MultiTrajectory fittedStates;
+  traj_t fittedStates;
 
   // These is used internally to store candidate trackstates
-  MultiTrajectory stateBuffer;
-  std::vector<MultiTrajectory::TrackStateProxy> trackStateCandidates;
+  traj_t stateBuffer;
+  std::vector<typename MultiTrajectory<traj_t>::TrackStateProxy>
+      trackStateCandidates;
 
   // This is the indices of the 'tip' of the tracks stored in multitrajectory.
   // This correspond to the last measurment state in the multitrajectory.
-  std::vector<size_t> lastMeasurementIndices;
+  std::vector<MultiTrajectoryTraits::IndexType> lastMeasurementIndices;
 
   // This is the indices of the 'tip' of the tracks stored in multitrajectory.
   // This correspond to the last state in the multitrajectory.
-  std::vector<size_t> lastTrackIndices;
+  std::vector<MultiTrajectoryTraits::IndexType> lastTrackIndices;
```

Comment:
In principle, yes. I added `MultiTrajectoryTraits` so you don't have to go throught `MultiTrajectory<T>` which is a template. I think in this case here I could use it, because I have the concrete type, but thought this is cleaner.